### DISCUSSION TITLE
Umami across subdomains

### DISCUSSION THREAD

**Discussion by pwbriggs** [umami-software/umami#2869] [Q&A]
I'm having trouble setting up Umami with subdomains. Here's my scenario:

At `mydomain.com`, I serve my product website that advertises my service. This site includes my marketing content and ad landing pages, etc. However, serve my actual webapp over at `app.mydomain.com`. I want to track users across the boundary between my app and my website, so they both use the same website ID for Umami.

As far as I can tell, though, Umami doesn't differentiate between `https://mydomain.com/`  and `https://app.mydomain.com/`&mdash;they both come out as `/`.

What am I missing here? Do I need to create a custom tracker function for this usecase, sending a custom value for `url` that includes the subdomain? Or is this an unsupported setup?

Thanks! ❤️

**Comment by pixelpaulaus** [comment]
i would also like this.

**Reply by ericlamotte** [reply]
I'm also interested in learning what the best practice is for this.

**Reply by 3nth** [reply]
You can filter the dashboards by hostname, but it seems sessions are not tracked across subdomains with the same websiteID.

SessionID includes `hostname` taken from `window.location.hostname` so each subdomain will be a separate session.
https://github.com/umami-software/umami/blob/d8e46e487c39a7ab5a9af89a869f79167b09c544/src/lib/session.ts#L42

**Reply by pwbriggs** [reply]
Right, @3nth, so that means we can't currently use things like the funnel report for this setup.

**Reply by mikecao** [reply]
We've removed hostname from session calculation so now you can track sessions across subdomains.

**Reply by pwbriggs** [reply]
@mikecao IIRC it wasn't session calculation that was the issue, it's that Umami's data analysis, such as the funnel report, treated `https://example.com/my-path` and `https://subdomain.example.com/my-path` as the same page.

**Reply by mikecao** [reply]
@pwbriggs If you are using the same website ID for both the domain and subdomain, yes it will aggregate under the same ID. But we have added support for filtering by hostname.

**Reply by pwbriggs** [reply]
@mikecao my main usecase for cross-domain tracking would be tracking how users get from `https://example.com/ad-landing-page` to `https://app.example.com/conversion`—for example by using the "funnel" query. As far as I can tell, I can only *filter* by host, whereas I think it would make more sense for reports to take host into account as part of the page identifier.

**Comment by pwbriggs** [comment]
I ended up having to work around this by restructuring my whole site. 😞
